packages/brokeneck-react/src/components/Entities.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react'
+import T from 'prop-types'
+import {
+  Box,
+  Button,
+  CircularProgress,
+  IconButton,
+  Link,
+  makeStyles,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography
+} from '@material-ui/core'
+import { Link as RouterLink, useRouteMatch } from 'react-router-dom'
+import { useQuery } from 'graphql-hooks'
+import startCase from 'lodash.startcase'
+
+import useCreateUserDialog from '../hooks/useCreateUserDialog'
+import useFields, { TYPE_NAMES } from '../hooks/useFields'
+import usePagination from '../hooks/usePagination'
+import useSearch from '../hooks/useSearch'
+
+import Square from './Square'
+
+const useStyles = makeStyles(theme => ({
+  spacing: {
+    display: 'flex',
+    alignItems: 'center',
+    '& > * + *': {
+      marginLeft: theme.spacing(2)
+    }
+  },
+  right: {
+    marginLeft: 'auto'
+  }
+}))
+
+export default function Entities({
+  query,
+  entityName,
+  entitiesName,
+  entitiesKey
+}) {
+  const match = useRouteMatch()
+  const classes = useStyles()
+  const fields = useFields(entityName)
+  const { search, Search } = useSearch()
+
+  const {
+    pageSize,
+    currentToken,
+    useUpdateToken,
+    useTablePagination
+  } = usePagination({ pageSizeOptions: [2, 3, 4] }) // TODO: temp small page sizes for testing
+
+  const { data, loading, refetch: loadEntities } = useQuery(query(fields.all), {
+    variables: {
+      pageNumber: currentToken,
+      pageSize,
+      search
+    }
+  })
+
+  const [dialog, openDialog] = useCreateUserDialog(loadEntities)
```

Comment:
this needs to be either abstracted or injected from outside and it's user specific and this component is intended to be used for both users and groups